### ISSUE TITLE
API error on docker image build

### ISSUE THREAD

**Issue by arokem** [richford/cloudknot#143]
Might be related to #142 ? When I resolve that one, I run into the following error on knot initialization: 

```
---------------------------------------------------------------------------
APIError                                  Traceback (most recent call last)
<ipython-input-12-a2332bfe229e> in <module>()
      4                resource_type="SPOT",
      5                bid_percentage=50,
----> 6                pars_policies=('AmazonS3FullAccess',))

/home/ubuntu/.local/lib/python2.7/site-packages/cloudknot/cloudknot.pyc in __init__(self, name, pars, pars_policies, docker_image, base_image, func, image_script_path, image_work_dir, image_github_installs, username, repo_name, image_tags, job_definition_name, job_def_vcpus, memory, retries, compute_environment_name, instance_types, resource_type, min_vcpus, max_vcpus, desired_vcpus, image_id, ec2_key_pair, ce_tags, bid_percentage, job_queue_name, priority)
   1374 
   1375             self._docker_image, self._docker_repo, repo_cleanup = \
-> 1376                 futures['docker-image'].result()
   1377 
   1378             futures['job-definition'] = executor.submit(

/home/ubuntu/.local/lib/python2.7/site-packages/concurrent/futures/_base.pyc in result(self, timeout)
    453                 raise CancelledError()
    454             elif self._state == FINISHED:
--> 455                 return self.__get_result()
    456 
    457             self._condition.wait(timeout)

/home/ubuntu/.local/lib/python2.7/site-packages/concurrent/futures/thread.pyc in run(self)
     61 
     62         try:
---> 63             result = self.fn(*self.args, **self.kwargs)
     64         except:
     65             e, tb = sys.exc_info()[1:]

/home/ubuntu/.local/lib/python2.7/site-packages/cloudknot/cloudknot.pyc in set_dockerimage(knot_name, input_docker_image, func, script_path, work_dir, base_image, github_installs, username, tags, repo_name)
   1053 
   1054                 if not di.images:
-> 1055                     di.build(tags=tags)
   1056                     mod_logger.info(
   1057                         'knot {name:s} built docker image {i!s}'

/home/ubuntu/.local/lib/python2.7/site-packages/cloudknot/dockerimage.pyc in build(self, tags, image_name)
    491                 path=self.build_path,
    492                 dockerfile=self.docker_path,
--> 493                 tag=im['name'] + ':' + im['tag']
    494             )
    495 

/home/ubuntu/.local/lib/python2.7/site-packages/docker/models/images.pyc in build(self, **kwargs)
    185         image_id = None
    186         result_stream, internal_stream = itertools.tee(json_stream(resp))
--> 187         for chunk in internal_stream:
    188             if 'error' in chunk:
    189                 raise BuildError(chunk['error'], result_stream)

/home/ubuntu/.local/lib/python2.7/site-packages/docker/utils/json_stream.pyc in split_buffer(stream, splitter, decoder)
     64     buffered = six.text_type('')
     65 
---> 66     for data in stream_as_text(stream):
     67         buffered += data
     68         while True:

/home/ubuntu/.local/lib/python2.7/site-packages/docker/utils/json_stream.pyc in stream_as_text(stream)
     20     instead of byte streams.
     21     """
---> 22     for data in stream:
     23         if not isinstance(data, six.text_type):
     24             data = data.decode('utf-8', 'replace')

/home/ubuntu/.local/lib/python2.7/site-packages/docker/api/client.pyc in _stream_helper(self, response, decode)
    314             # Response isn't chunked, meaning we probably
    315             # encountered an error immediately
--> 316             yield self._result(response, json=decode)
    317 
    318     def _multiplexed_buffer_helper(self, response):

/home/ubuntu/.local/lib/python2.7/site-packages/docker/api/client.pyc in _result(self, response, json, binary)
    229     def _result(self, response, json=False, binary=False):
    230         assert not (json and binary)
--> 231         self._raise_for_status(response)
    232 
    233         if json:

/home/ubuntu/.local/lib/python2.7/site-packages/docker/api/client.pyc in _raise_for_status(self, response)
    225             response.raise_for_status()
    226         except requests.exceptions.HTTPError as e:
--> 227             raise create_api_error_from_http_exception(e)
    228 
    229     def _result(self, response, json=False, binary=False):

/home/ubuntu/.local/lib/python2.7/site-packages/docker/errors.pyc in create_api_error_from_http_exception(e)
     29         else:
     30             cls = NotFound
---> 31     raise cls(e, response=response, explanation=explanation)
     32 
     33 

APIError: 500 Server Error: Internal Server Error ("Cannot locate specified Dockerfile: /home/ubuntu/analysis/cloudknot_docker_color_fa_uadyAR/Dockerfile")

```

Have you seen this one before?

**Comment by richford** [comment]
So docker-py upgraded with backward-compatibility breaking API enhancements. I tried to track down the change but wound up simply specifying `docker>=2.0.0, <3.0.0` in `version.py` and `requirements.txt`. Should be fixed with #142.

**Comment by richford** [comment]
Resolved in #142
